### PR TITLE
chore: cl unbonding positions query

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

`/osmosis/concentratedliquidity/v1beta1/user_unbonding_positions/{address}`

Adds a query to list all unbonding positions of a given address, which also includes the lock information for that position

### CODE REVIEW

File: proto/osmosis/concentrated-liquidity/query.proto [hunk 1]
```diff
@@ -108,6 +108,13 @@ service Query {
                                    "cfmm_pool_id_link_from_concentrated/"
                                    "{concentrated_pool_id}";
   }
+
+  rpc UserUnbondingPositions(UserUnbondingPositionsRequest)
+      returns (UserUnbondingPositionsResponse) {
+    option (google.api.http).get = "/osmosis/concentratedliquidity/v1beta1/"
+                                   "user_unbonding_positions/"
+                                   "{address}";
```
Comment (p0mvn) [inline proto/osmosis/concentrated-liquidity/query.proto:116]: AFAIR, specifying this on separate lines causes issues. CC: @mattverse for confirmation

File: x/concentrated-liquidity/pool.go [hunk 2]
```diff
@@ -340,3 +343,45 @@ func validateAuthorizedQuoteDenoms(ctx sdk.Context, denom1 string, authorizedQuo
...
+				continue
+			}
+
+			userPositionsWithPeriodLocks = append(userPositionsWithPeriodLocks, model.PositionWithPeriodLock{
+				Position: pos,
+				Locks:    *lock,
+			})
+		default:
+			continue
+		}
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool.go:384]: Similar to the comment in another PR, why do we have a default case here?

File: x/concentrated-liquidity/pool.go [hunk 3]
```diff
@@ -340,3 +343,45 @@ func validateAuthorizedQuoteDenoms(ctx sdk.Context, denom1 string, authorizedQuo
...
+	for _, pos := range positions {
+		lockId, err := k.GetLockIdFromPositionId(ctx, pos.PositionId)
+		switch err.(type) {
+		case types.PositionIdToLockNotFoundError:
+			continue
+		case nil:
+			// If we have hit this logic branch, it means that, at one point, the lockId provided existed. If we fetch it again
+			// and it doesn't exist, that means that the lock has matured.
+			lock, err := k.lockupKeeper.GetLockByID(ctx, lockId)
+			if err == errorsmod.Wrap(lockuptypes.ErrLockupNotFound, fmt.Sprintf("lock with ID %d does not exist", lock.GetID())) {
```
Comment (p0mvn) [inline x/concentrated-liquidity/pool.go:366]: Can this be compared with `errors.Is`?

**Review by p0mvn — APPROVED** [review]

File: proto/osmosis/concentrated-liquidity/query.proto (hunk 1 above)
Comment (czarcas7ic) [reply proto/osmosis/concentrated-liquidity/query.proto:116]: There is no way to do that with the linter set right?

File: x/concentrated-liquidity/pool.go (hunk 2 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/pool.go]: https://github.com/osmosis-labs/osmosis/pull/5684/commits/6715ba8e5876cad3229e030e12dcfc27dd75ba63

File: x/concentrated-liquidity/pool.go (hunk 3 above)
Comment (czarcas7ic) [reply x/concentrated-liquidity/pool.go]: https://github.com/osmosis-labs/osmosis/pull/5684/commits/6715ba8e5876cad3229e030e12dcfc27dd75ba63

File: proto/osmosis/concentrated-liquidity/query.proto (hunk 1 above)
Comment (czarcas7ic) [reply proto/osmosis/concentrated-liquidity/query.proto:116]: Jon has been utilizing this endpoint locally, so I am going to assume its good to go. Going to merge this to reduce overhead of open PRs

**Pull request merged**